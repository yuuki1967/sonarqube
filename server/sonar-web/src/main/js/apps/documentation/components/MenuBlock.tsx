/*
 * SonarQube
 * Copyright (C) 2009-2019 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import * as React from 'react';
import { MenuItem } from './MenuItem';
import { DocumentationEntry, DocsNavigationBlock, getNodeFromUrl } from '../utils';
import OpenCloseIcon from '../../../components/icons-components/OpenCloseIcon';

interface Props {
  block: DocsNavigationBlock;
  onToggle: (title: string) => void;
  open: boolean;
  pages: DocumentationEntry[];
  splat: string;
  title: string;
}

export default class MenuBlock extends React.PureComponent<Props> {
  handleClick = (event: React.MouseEvent<HTMLAnchorElement>) => {
    event.stopPropagation();
    event.preventDefault();
    this.props.onToggle(this.props.title);
  };

  render() {
    const { open, block, pages, title, splat } = this.props;
    return (
      <>
        <a className="list-group-item" href="#" onClick={this.handleClick}>
          <h3 className="list-group-item-heading">
            <OpenCloseIcon className="little-spacer-right" open={this.props.open} />
            {title}
          </h3>
        </a>
        {open &&
          block.children.map(item => (
            <MenuItem indent={true} key={item} node={getNodeFromUrl(pages, item)} splat={splat} />
          ))}
      </>
    );
  }
}
