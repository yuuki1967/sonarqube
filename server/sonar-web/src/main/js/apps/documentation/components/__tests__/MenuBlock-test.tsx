/*
 * SonarQube
 * Copyright (C) 2009-2019 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import * as React from 'react';
import { shallow } from 'enzyme';
import MenuBlock from '../MenuBlock';

const block = {
  title: 'Foo',
  children: ['/bar/', '/baz/']
};

const pages = [
  {
    content: 'bar',
    relativeName: '/bar/',
    text: 'bar',
    title: 'Bar',
    navTitle: undefined,
    url: '/bar/'
  },
  {
    content: 'baz',
    relativeName: '/baz/',
    text: 'baz',
    title: 'baz',
    navTitle: 'baznav',
    url: '/baz/'
  }
];

it('should render a closed menu block', () => {
  expect(
    shallow(
      <MenuBlock
        block={block}
        onToggle={jest.fn()}
        open={false}
        pages={pages}
        splat="/foobar/"
        title="Foobarbaz"
      />
    )
  ).toMatchSnapshot();
});

it('should render an opened menu block', () => {
  expect(
    shallow(
      <MenuBlock
        block={block}
        onToggle={jest.fn()}
        open={true}
        pages={pages}
        splat="/foo/"
        title="Foo"
      />
    )
  ).toMatchSnapshot();
});
