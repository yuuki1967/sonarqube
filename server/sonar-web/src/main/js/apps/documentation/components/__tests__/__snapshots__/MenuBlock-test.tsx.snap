// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a closed menu block 1`] = `
<Fragment>
  <a
    className="list-group-item"
    href="#"
    onClick={[Function]}
  >
    <h3
      className="list-group-item-heading"
    >
      <OpenCloseIcon
        className="little-spacer-right"
        open={false}
      />
      Foobarbaz
    </h3>
  </a>
</Fragment>
`;

exports[`should render an opened menu block 1`] = `
<Fragment>
  <a
    className="list-group-item"
    href="#"
    onClick={[Function]}
  >
    <h3
      className="list-group-item-heading"
    >
      <OpenCloseIcon
        className="little-spacer-right"
        open={true}
      />
      Foo
    </h3>
  </a>
  <MenuItem
    indent={true}
    key="/bar/"
    node={
      Object {
        "content": "bar",
        "navTitle": undefined,
        "relativeName": "/bar/",
        "text": "bar",
        "title": "Bar",
        "url": "/bar/",
      }
    }
    splat="/foo/"
  />
  <MenuItem
    indent={true}
    key="/baz/"
    node={
      Object {
        "content": "baz",
        "navTitle": "baznav",
        "relativeName": "/baz/",
        "text": "baz",
        "title": "baz",
        "url": "/baz/",
      }
    }
    splat="/foo/"
  />
</Fragment>
`;
